packages/mermaid/src/themes/theme-default.js

```diff
@@ -272,6 +272,23 @@ class Theme {
       this.quadrantExternalBorderStrokeFill || this.primaryBorderColor;
     this.quadrantTitleFill = this.quadrantTitleFill || this.primaryTextColor;
 
+    /* xychart */
+    this.xyChart = {
+      backgroundColor: this.xyChart?.backgroundColor || this.background,
+      titleColor: this.xyChart?.titleColor || this.primaryTextColor,
+      xAxisTitleColor: this.xyChart?.xAxisTitleColor || this.primaryTextColor,
+      xAxisLabelColor: this.xyChart?.xAxisLabelColor || this.primaryTextColor,
+      xAxisTickColor: this.xyChart?.xAxisTickColor || this.primaryTextColor,
+      xAxisLineColor: this.xyChart?.xAxisLineColor || this.primaryTextColor,
+      yAxisTitleColor: this.xyChart?.yAxisTitleColor || this.primaryTextColor,
+      yAxisLabelColor: this.xyChart?.yAxisLabelColor || this.primaryTextColor,
+      yAxisTickColor: this.xyChart?.yAxisTickColor || this.primaryTextColor,
+      yAxisLineColor: this.xyChart?.yAxisLineColor || this.primaryTextColor,
```

Comment:
It would be better but it requires lots of changes so want to confirm if we are going with the changes? as all other diagrams have linear structures. @sidharthv96 @nirname

Reply:
I do not actually know who is responsible for styled-diagrams approach, perhaps we should ask someone else from the team. I am OK with both options, flattened or nested. Since there are only 2 axis, X and Y, there is no point in redoing all the job solely for scaling purposes. In the worst case we will add Z-axis and some 3d.

By using `-beta` suffix we reserve the right to make breaking changes. I would say that we can postpone this decision till the time we remove this `-beta` option. Right now lets keep it similar to the other diagrams. Consistency is important. Since everything is flatten, let this be flatten too.